akka-http2-support/src/main/scala/akka/http/impl/engine/http2/framing/Http2FrameParsing.scala

```diff
@@ -162,5 +182,28 @@ private[http2] class Http2FrameParsing(shouldReadPreface: Boolean) extends ByteS
             UnknownFrameEvent(tpe, flags, streamId, payload.remainingData)
         }
       }
+
+      // FIXME this should be smarter, we need to somehow manage when we send back the ACK basically...
+      private def applySetting(s: SettingsFrame): SettingsFrame = {
+        s.settings.foreach {
+          case Setting(SettingIdentifier.SETTINGS_MAX_FRAME_SIZE, value) ⇒
+            settings.updateMaxOutFrameSize(value)
```

Comment:
I think this logic might actually be wrong. As discussed in https://github.com/akka/akka-http/pull/764#issuecomment-272897513 a received SETTINGS_MAX_FRAME_SIZE doesn't say something about the frames that our peer sends (and we receive here) but about the frames which our peer receives:

```
  SETTINGS_MAX_FRAME_SIZE (0x5):  Indicates the size of the largest
      frame payload that the sender is willing to receive, in octets.
```

so, as long as we ourselves don't change this setting, we cannot act on it here (and if, we would have to act on the settings ACK frame instead).

It's actually the same for `SETTINGS_ENABLE_PUSH`, so we might not need any handling of settings frames in the parser right now (apart from checking the limits against the default values.)